cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconcilerStatusTest.java

```diff
@@ -900,5 +897,43 @@ protected Future<Void> listeners()  {
 
             return Future.succeededFuture();
         }
+
+        @Override
+        protected Future<Void> initClientAuthenticationCertificates() {
+            kafkaTlsPemIdentity = new TlsPemIdentity(null, null);
+            return Future.succeededFuture();
+        }
+    }
+
+    static class MockKafkaReconcilerFailsWithVersionUpdate extends KafkaReconciler {
```

Comment:
Why is this change needed? It does not seem related to the PR itself.

Reply:
Previously the test testKafkaReconcilerStatusDoesNotUpdateVersionOnFailure relied on the clusterId step in the reconciliation failing due to the certificate secrets being missing, so that the reconciliation loop would fail and the test could check a version update did not occur. Now with the certificates being handled differently the mocked secrets call it was relying on doesn't cause a failure. So I updated the test to explicitly add a step that failed. This makes the test less brittle and focuses it on testing the behaviour it's supposed to be testing.

Reply:
I don't get this -> the missing secret should still throw some error, or? Or why does it not fail? That seems like a missing validation?

Reply:
I updated the test class so that instead of mocking the secret everywhere I provided an override of the `initClientAuthenticationCertificates` that initializes the PemIdentity variable and returns a SucceededFuture. So for the test `testKafkaReconcilerStatusDoesNotUpdateVersionOnFailure` that expects a failure the equivalent change would be to update MockKafkaReconcilerStatusTasks to have an option where the `initClientAuthenticationCertificates` method fails. However, since the test doesn't really have anything to do with `initClientAuthenticationCertificates` and instead is just testing that some failure somewhere in the reconcile loop means the UpdateVersion does not happen it seemed sensible to decouple the two and instead provide a separate MockKafkaReconciler that has an explicit step in the reconcile loop that fails.

Reply:
It looks like I actually forgot to take out the other unnecessary secret mocks from this class, so I'll push up an extra commit to remove them

Reply:
Ok, that makes sense I guess. Thanks.

Reply:
> It looks like I actually forgot to take out the other unnecessary secret mocks from this class, so I'll push up an extra commit to remove them

Too late. Sorry :-(